include/aspect/simulator.h

```diff
@@ -1170,6 +1170,26 @@ namespace aspect
       void output_statistics();
 
       /**
+       * This function is called at the end of each time step and generates
+       * global statistics about this time step like the current time, the
+       * number of mesh cells, and the number of degrees of freedom.
+       *
+       * In addition, it is called at the beginning of the model run to add the
+       * columns to the statistics file in the correct order. In case that
+       * only one nonlinear iteration will be made, it generates output for
+       * the first row of the statistics file and returns true. If the first
+       * time step has more than one nonlinear iteration, it fills the
+       * respective entries with zeroes and returns false.
+       *
+       * The input argument 'statistics_already_generated' determines which
+       * of the two cases applies.
```

Comment:
do you mean to say that the question above "In case that only one nonlinear iteration will be made" is not determined by the called function, but by this flag?

If so, above maybe say "If @p statistics_already_generated is false, then ... do this and that ... . This function argument is typically set to @p true at the calling site if ... and to @p false if ... ."